Celeste.Mod.mm/Patches/Monocle/Entity.cs

```diff
@@ -13,5 +15,26 @@ class patch_Entity : Entity {
         internal void DissociateFromScene() {
             Scene = null;
         }
+
+        public List<Action> UpdatePrecederActions;
+        public List<Action> UpdateFinalizerActions;
+
+        //For some reason this compiles to be significantly slower than it could be in theory, most likely to do patch structure in MonoMod.
+        //If you want me to update this to be written in pure IL I can do that as well, just let me know in PR review
```

Comment:
Please elaborate in what you mean by those comments. Specifically, I'm interested in how fast it should be in theory and how MonoMod is holding you back here.

Please keep in mind that C# / IL that looks "less optimized" can still be JIT-compiled to native code that's more optimized than the "optimized C#" counterpart in certain instances. Also, please stop hyperfocusing on micro-optimizations in a community-maintained project that you yourself agreed to that it deserves to be rewritten from scratch as there are many more macro-optimizations and larger problems on the table. Sometimes code clarity and having code that's maintainable by as many people as possible is much more important than writing code that's supposedly faster (I'd like to see benchmarks) with a less clear intent behind why it's done that way and that might even end up misleading new programmers stumbling upon such arcane practices.

Lastly, why not use events with the delegate += / -= syntax?

Reply:
I am uncertain whether or not an instance `event` will be slower than this methodology. I had asked about it in #modding-irl with no response so I ran with this originally.

Also my compiles were different in a few cases, some of which had no-ops littered in with no real context, I assume this is a MonoMod thing but my thought process is if this is happening per entity per update, I'd want to optimize this as much as possible.